### ISSUE TITLE
Unexpected CS8907 on record with parameter of that record type

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#49628]
Reported by [controlflow](https://twitter.com/controlflow/status/1331734923850485768)
```
record Record(Record Inner);
// CS8907: Parameter 'Inner' is unread. Did you forget to use it to initialize the property with that name?
// why generated constructor is empty?

namespace System.Runtime.CompilerServices {
  public class IsExternalInit { }
}
```

**Comment by Youssef1313** [comment]
Looks like a regression. I tested on an old branch, and the generated constructor is correct.

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRIMYHsEBNkA+mOuABAEoAU5pAkgHb0QICUA3ALABQA9D6QGEAygA4AnAAYA7CFIAFKAigBbCDGakA5AyYJNpAJbJScekii4AdKQAiBsgE8scUmBwBzNaRhYTyCIYw3r4G9AYwBlAANgYAXgEwABYBAA4IWCnMMA6kAO7hid6JsKT0KhAA/Nx8eYk5nrqwEGTY9MgwCHAYPgiGxhDKKdlVXNxlqsgpUBgBAAIAjJbkphGqlgJYgwZRzELMAG4GM8YA3tykpLMAzJcATHTIAKIAHuoIZVEM4aQnpAC+3ABXCAA

**Comment by jcouv** [comment]
Merged fix for 16.8p3
